src/realm/db.hpp

```diff
@@ -439,7 +438,7 @@ class DB : public std::enable_shared_from_this<DB> {
     class ReadLockGuard;
 
     // Member variables
-    std::recursive_mutex m_mutex;
+    mutable std::recursive_mutex m_mutex;
```

Comment:
A recursive lock means we expect the lock to be re-taken by the same thread while it is holding it. Is this because this `mutex`  is part of some recursive code?

Reply:
Or just that one function acquiring the mutex will call another function that will also try to acquire it.